xarray/core/indexes.py

```diff
@@ -193,7 +307,104 @@ def copy(self, deep: bool = True) -> "PandasIndex":
         # >>> len(pickle.dumps((self.array, self.array.copy(deep=False))))
         # 8000341
         array = self.array.copy(deep=True) if deep else self.array
-        return PandasIndex(array, self._dtype)
+        return type(self)(array, self._dtype)
+
+
+class PandasMultiIndex(PandasIndex):
+    def query(
+        self, labels, method=None, tolerance=None
+    ) -> Tuple[Any, Union["PandasIndex", None]]:
```

Comment:
If others who've been following this closer than me understand then great — I don't want to slow us down. 

I don't quite understand the role of returning another index — to the extent that's easy to explain, that would be helpful. Thanks!

Reply:
No worries. It's useful for example in the case of `pandas.MultiIndex` where `get_loc_level` might drop index levels and return a simple `pd.Index` that we can then reuse to construct the result Xarray object. Maybe it would be useful for other, custom indexes too?

Reply:
This would be great for a "staggered grid index" like "XgcmIndex" (https://github.com/xgcm/xgcm/issues/200)

Reply:
Thanks. I probably need to do some more work here if I'm going to be helpful. Thanks for the link.